src/main/java/ikysil/training/dao/impl/InMemoryOrderDao.java

```diff
@@ -37,7 +37,14 @@ public OrderDto updateOrder(String id, OrderDto order) {
     }
 
     @Override
-    public void deleteOrder(String id) {
+    public OrderDto deleteOrder(String id) {
+        OrderDto order = storage.get(id);
         storage.remove(id);
```

Comment:
if you read java doc for `remove` method:
"the previous value associated with key, or null if there was no mapping for key."
so do just:
```
return storage.remove(id);
```